tmt/base.py

```diff
@@ -465,13 +468,26 @@ def __init__(self, node, run=None):
     @property
     def environment(self):
         """ Return combined environment from plan data and command line """
-        if self.run and self.run.environment:
+        if self.my_run and self.my_run.environment:
             combined = self._environment.copy()
-            combined.update(self.run.environment)
+            combined.update(self.my_run.environment)
             return combined
         else:
             return self._environment
 
+    def _initialize_workdir_tree(self):
+        """
+        Workdir tree, a copy of the tree root, used as cwd in prepare, execute
+        and finish steps
+        """
+        # Bail out if workdir does not exist, e.g. not running the plan ...
+        if not self.workdir:
+            return
+
+        self.workdir_tree = os.path.join(self.workdir, 'tree')
+        tree_root = self.my_run.tree.root or '.'
```

Comment:
Why defaulting to `.`? There has to be a metadata tree root. Otherwise we would copy a random current working directory? Or did I miss anything?

Reply:
hmm, trying to remember :)

Reply:
@psss maybe bacause of this?

```
$ tmt
No metadata found in the '.' directory. Use 'tmt init' to get started.

$ tmt run -a provision -h container execute --how shell --script "echo hi"
/var/tmp/tmt/run-015

/plans/default
    warn: The 'shell' execute method has been deprecated.
    warn: Use 'how: tmt' in the execute step instead (L2).
    warn: Set 'framework: shell' in test metadata (L1).
    warn: Support for old methods will be dropped in March 2021.
    discover
        how: shell
        summary: 1 test selected
    provision
        how: container
        image: fedora
        distro: Fedora 33 (Container Image)
        summary: 1 guest provisioned
    prepare
        summary: 0 preparations applied
    execute
        how: tmt
        summary: 1 test executed
    report
        how: display
        summary: 1 test passed
    finish
        container: stopped
        container: removed
        summary: 0 tasks completed

total: 1 test passed
```

Reply:
I think this could be quite dangerous: Let's say I just want to reserve a box for experimenting using:

    tmt run provision login finish

And because I just started a new shell for that I'm standing in my home directory. So this would start rsyncing my whole home directory to the workdir...

Reply:
Agreed to skip copy in this case, and jsut print a debug message.